src/eval.c

```diff
@@ -183,14 +235,17 @@ int luaRedisReplicateCommandsCommand(lua_State *lua) {
  *
  * However it is simpler to just call scriptingReset() that does just that. */
 void scriptingInit(int setup) {
-    lua_State *lua = lua_open();
```

Comment:
i suddenly realized that this change doesn't cover functions.
i.e. they have a separate call to lua_open, and we'll probably want both Lua VMs to share an arena and a tcache.

Reply:
yes, i want to handle function after the eval lua is completely ok...

Reply:
@oranagra now we're just sharing the areana between lua VM(script and function), not tcache.
this is because when free the lua vm async, we have to avoid both the bio and the main thread using the tcache at the same time.

Reply:
ok, i now notice this is explained in the top comment. so all is good?